### ISSUE TITLE
False positive for MA0105

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#569]
### Version of the Meziantou.Analyzer NuGet package

2.0.68

### Rule Identifier

MA0105

### Target Framework

netstandard2.0

### C# Language version

C# 12

### Description

MA0105 is reported incorrectly when `addValue` is used inside `updateValueFactory`

### Reproduction Steps

Minimal code:

```c#
concurrentDictionary.AddOrUpdate(name, newValue, (_, oldValue) => behavior switch
{
    Behavior.Ignore => oldValue,
    Behavior.Append => $"{oldValue}, {newValue}",
    Behavior.Override => newValue,
    _ => throw new ArgumentOutOfRangeException(nameof(behavior), behavior, "Unknown behavior."),
}); <- says I capture newValue, but I don't have an option not to capture it.

// btw there's no such FP for behavior
```

**Comment by meziantou** [comment]
````c#
ConcurrentDictionary<string, string> concurrentDictionary = new(StringComparer.Ordinal);
var name = "";
var newValue = "";
var behavior = Behavior.Ignore;

concurrentDictionary.AddOrUpdate(name, (_, arg) => arg.newValue, (_, oldValue, arg) => arg.behavior switch
{
    Behavior.Ignore => oldValue,
    Behavior.Append => $"{oldValue}, {arg.newValue}",
    Behavior.Override => arg.newValue,
    _ => throw new ArgumentOutOfRangeException(nameof(behavior), arg.behavior, "Unknown behavior."),
}, (newValue, behavior));

enum Behavior
{
    Ignore,
    Append,
    Override,
}
````

**Comment by hankovich** [comment]
It's `netstandard2.0`.

And if this overload is available, then MA0105 should also be reported for `behavior`.

**Comment by meziantou** [comment]
It will be fixed in the next version
